loki/transformations/single_column/devector.py

```diff
@@ -0,0 +1,248 @@
+# (C) Copyright 2018- ECMWF.
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+from more_itertools import split_at
+
+from loki.analyse import dataflow_analysis_attached
+from loki.batch import Transformation
+from loki.ir import (
+    nodes as ir, FindNodes, FindScopes, FindVariables, Transformer,
+    NestedTransformer, is_loki_pragma, pragmas_attached,
+)
+from loki.tools import as_tuple, flatten
+from loki.types import BasicType
+
+from loki.transformations.utilities import (
+    find_driver_loops, check_routine_sequential
+)
+
+
+__all__ = [
+    'RemoveLoopTransformer', 'SCCDevectorTransformation',
+]
+
+
+class RemoveLoopTransformer(Transformer):
+    """
+    A :any:`Transformer` that removes all loops over the specified
+    dimension.
+
+    Parameters
+    ----------
+    horizontal : :any:`Dimension`
+        The dimension specifying the horizontal vector dimension
+    """
+    # pylint: disable=unused-argument
+
+    def __init__(self, dimension, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.dimension = dimension
+
+    def visit_Loop(self, loop, **kwargs):
+        if loop.variable == self.dimension.index:
+            # Recurse and return body as replacement
+            return self.visit(loop.body)
```

Comment:
```suggestion
            return self.visit(loop.body, **kwargs)
```
Otherwise any derived implementations that may want to utilitze additional kwargs won't be able to pass them through here.